common/server.ts

```diff
@@ -1,61 +1,63 @@
 import axios from "axios";
+import { updateToken } from "../stores/slices/userSlice";
 import localStorage from "./localStorage";
 
-const handleUnauthorized = () => {//TODO: CHANGES FOR TOKEN AND LOCATION
-    const { pathname, search } = window.location;
-    //removeAccessToken();
-    const currentUrl = encodeURIComponent(pathname.concat(search));
-    //window.location.href = `${ROUTES.login}?redirect=${currentUrl}`;
+const handleUnauthorized = async (dispatch) => {
+    await localStorage.removeItem('token');
+    dispatch?.(updateToken(''))
   };
 
-const handleResponse = (response) => {
-    if (response.error) {
-      if (response.status === 401) {
-        handleUnauthorized();
-      }
-  
-      const errorMessage = response.data?.message || response.statusText;
-      console.error(errorMessage);
-      // TODO: Give visual feedback
-    }
-  
-    return response;
-  };
-  
-  const handleError = (error) => {
-    if (error.response?.status === 401) {
-      console.log('Not authorized for execute this operation (401)');
-      handleUnauthorized();
-    }
-    return {
-      status: error.response?.status,
-      message: error.response?.data?.message,
-    };
-  };
-  
-  const request = async (url: string, data?, method: string = 'GET', options = {}) => {
-    const authToken = await localStorage.getItem('token')
-    const requestOptions = {
-      url,
-      method,
-      data,
-      headers: {},
-    };
-  
-    if (authToken) {
-      requestOptions.headers.Authorization = `Bearer ${authToken}`;
+const handleResponse = (response, dispatch) => {
+  if (response.error) {
+    if (response.status === 401) {
+      handleUnauthorized(dispatch);
     }
-  
-    return axios({ ...requestOptions, ...options })
-      .then(handleResponse)
-      .catch(handleError);
+
+    const errorMessage = response.data?.message || response.statusText;
+    console.error(errorMessage);
+    // TODO: Give visual feedback
+  }
+
+  return response;
+};
+  
+const handleError = (error, dispatch) => {
+  if (error.response?.status === 401) {
+    console.log('Not authorized for execute this operation (401)');
+    handleUnauthorized(dispatch);
+  }
+  return {
+    status: error.response?.status,
+    message: error.response?.data?.message,
```

Comment:
@KariGarcia just a question for me to understand. Here if the data is undefined what is assigned to the message?

Reply:
nothing 😆, setting a default message